### DISCUSSION TITLE
VSCode Source Control commit trailers not working properly

### DISCUSSION THREAD

**Discussion by krupis** [microsoft/vscode-discussions#986] [Extension Development QnA] [answered]
Hello. I am using VScode Source Control feature to stage, commit and push my changes to the linked repository.
I am also using Gitlab Changelog API to generate changelog based on commit titles and trailers. In order for the Gitlab Changelog to work, all important commits must have a trailer, for example commiting using CLI:
```
git commit -m "commiting with trailer" --trailer "Changelog: added"
```

I could not find a way to add trailers in VSCode Source Control. I have initially thought that a trailer is just part of the commit so I have tried the following:
![image](https://github.com/microsoft/vscode-discussions/assets/22237034/973ce08d-f2dc-4dd4-b4a9-44a44c044227)

When checking this on the repository, everything looks ok:
![image](https://github.com/microsoft/vscode-discussions/assets/22237034/c07fd057-a0c4-4948-b8c7-f0cc7186290b)

But when trying to generate a changelog, it does not seem to detect the commits that were performed using Source Control but it can detect commits that were performed using Fork and CLI.

```
PS C:\Users\petrikas.lu\Desktop\WORK\gitchangelog-test\my_project> glab changelog generate
## 3.5.0-19-g5a89eda
 (2023-12-20)

### added

- [Another commit using fork](embeded-programmers/gitchangelog-test@9889ae912d162ca546813aea108a531c4ebfa2c5) ()

- [Added feature using Fork Changelog: added](embeded-programmers/gitchangelog-test@faa951e4f07ef31d0a4abb419406f4c9aaed5a40) ()

- [Commit with trailer using command line](embeded-programmers/gitchangelog-test@d2bc0ad1d9ed32620de239e93170f426b251cfde) ()

```

I would appreciate if someone could shed some light on Source Control feature on VSCode and whether there is a possibility to add a trailer so the Gitlab Changelog API can detect it. Thanks in advance.

**Comment by gjsjohnmurray** [comment, marked as answer]
Are you **developing VS Code extensions**? If not, please redirect your question using the information in this article which is pinned at the top of this forum:

https://github.com/microsoft/vscode-discussions/discussions/5

**Reply by krupis** [reply]
You are right. I will try to post this somewhere else
